great_expectations/data_context/data_context/context_factory.py

```diff
@@ -96,7 +96,7 @@ def get_context(  # noqa: PLR0913
     cloud_organization_id: str | None = ...,
     cloud_mode: bool | None = ...,
     mode: None = ...,
-) -> AbstractDataContext:
+) -> EphemeralDataContext:
```

Comment:
```suggestion
) -> CloudDataContext | FileDataContext | EphemeralDataContext:
```

I think we actually need a 4th overload to capture just the `EphemeroralDataContext` case, the arguments above do not guarantee an `EphemoralContext` will be returned.

```python
@overload
def get_context(  # noqa: PLR0913
    project_config: DataContextConfig | Mapping | None = ...,
    context_root_dir: None = ...,
    project_root_dir: PathStr | None = ...,
    runtime_environment: dict | None = ...,
    cloud_base_url: None = ...,
    cloud_access_token: None = ...,
    cloud_organization_id: None = ...,
    cloud_mode: None = ...,
    mode: Literal["ephemeral"] = ...,
) -> EphemeralDataContext:

```